mso/resource_mso_schema_template_contract_service_graph.go

```diff
@@ -1085,229 +392,83 @@ func resourceTemplateContractServiceGraphRead(d *schema.ResourceData, m interfac
-			}
-
-			if siteMap["provider_subnet_ips"] != nil {
-				allMap["provider_subnet_ips"] = siteMap["provider_subnet_ips"]
-			}
-			if siteMap["consumer_subnet_ips"] != nil {
-				allMap["consumer_subnet_ips"] = siteMap["consumer_subnet_ips"]
-			}
-
 			nodeList = append(nodeList, allMap)
 		}
-		d.Set("schema_id", schemaID)
-		d.Set("site_id", siteID)
-		d.Set("template_name", templateName)
 		d.Set("node_relationship", nodeList)
-		d.Set("contract_name", contractName)
+		d.SetId(fmt.Sprintf("%s/templates/%s/contracts/%s", d.Get("schema_id").(string), templateName, contractName))
 
-		if d.Get("service_graph_name") == serviceGraph {
-			d.SetId(serviceGraph)
-		} else {
-			d.SetId("")
-		}
 	} else {
 		d.SetId("")
 	}
-
-	log.Printf("[DEBUG] Completed Read Template Contract Service Graph")
 	return nil
 }
 
-func resourceTemplateContractServiceGraphDelete(d *schema.ResourceData, m interface{}) error {
-	log.Printf("[DEBUG] Begining Delete Template Contract Service Graph")
+// PostSchemaTemplateContractServiceGraphConfig creates a new service graph configuration for a template contract.
+//
+// Parameters:
+//   - action: The action to perform.
+//   - d: The schema resource data.
+//   - m: The client interface.
+//
+// Returns:
+//   - An error if there was a problem creating the service graph configuration.
+func PostSchemaTemplateContractServiceGraphConfig(action string, d *schema.ResourceData, m interface{}) error {
 	msoClient := m.(*client.Client)
-
-	schemaID := d.Get("schema_id").(string)
-	siteID := d.Get("site_id").(string)
+	schemaId := d.Get("schema_id").(string)
 	TemplateName := d.Get("template_name").(string)
 	contractName := d.Get("contract_name").(string)
+	serviceGraph := d.Get("service_graph_name").(string)
 
-	tempPath := fmt.Sprintf("/templates/%s/contracts/%s/serviceGraphRelationship", TemplateName, contractName)
-	tempConGraph := models.NewTemplateContractServiceGraph("remove", tempPath, nil, nil)
-
-	sitePath := fmt.Sprintf("/sites/%s-%s/contracts/%s/serviceGraphRelationship", siteID, TemplateName, contractName)
-	siteConGraph := models.NewSiteContractServiceGraph("remove", sitePath, nil, nil)
-
-	response, err := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaID), tempConGraph, siteConGraph)
-
-	// Ignoring Error with code 141: Resource Not Found when deleting
-	if err != nil && !(response.Exists("code") && response.S("code").String() == "141") {
-		return err
-	}
-
-	d.SetId("")
-	log.Printf("[DEBUG] Completed Delete Template Contract Service Graph")
-	return nil
-}
+	serviceGraphRef := make(map[string]interface{})
+	serviceGraphRef["serviceGraphName"] = serviceGraph
 
-func getTemplateServiceGraph(cont *container.Container, templateName, graphName string) (*container.Container, int, error) {
-	templateCount, err := cont.ArrayCount("templates")
-	if err != nil {
-		return nil, -1, fmt.Errorf("No Template found")
+	if graphSchema, ok := d.GetOk("service_graph_schema_id"); ok {
```

Comment:
should be keep the naming aligned with the schema definition?
```suggestion
	if serviceGraphSchemaId, ok := d.GetOk("service_graph_schema_id"); ok {
```